lib/utils.js

```diff
@@ -55,7 +55,7 @@ function expandTabs(line) {
 exports.expandTabs = expandTabs;
 
 function parseOutput(lines, inputFilename, pathPrefix) {
-    const re = /^\s*<source>[:(]([0-9]+)(:?,?([0-9]+):?)?[):]*\s*(.*)/;
+    const re = /^(\w+)?.*<source>[: (]([0-9]+)(:?,?([0-9]+):?)?[:)]*\s*(.*)/;
```

Comment:
Just as a note: If the tests are passing, it might be nothing, but I'm concerned the first change from `\s` to `(\w+)?.` might not be compatible with all the different outputs from other compilers, but I have not been able to find an instance of said problem, so if anyone knows better and can find an example where it breaks the output parsing, please let us know.

(Also, I do prefer the happy face rather than the sad face 👍)

Reply:
I could make it more specific to prevent accidental matches.
That would make it something like: (not tested) 
`/^((?Warning|Error) [EW][0-9]{4})\s+<source>[(: ]`
I think that would be better anyway, since it would include the error code in the text